src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1617,13 +1617,30 @@ public static BigInteger Pow(BigInteger value, Int32 exponent)
         {
             left.AssertValid();
             right.AssertValid();
+            
+            var result = default(uint[]);
 
-            int sign = +1;
-            BigIntegerBuilder reg1 = new BigIntegerBuilder(left, ref sign);
-            BigIntegerBuilder reg2 = new BigIntegerBuilder(right, ref sign);
+            if (left._bits != null)
+            {
+                if (right._bits != null)
+                {
+                    if (left._bits == right._bits)
+                        result = BigCalc.Square(left._bits);
+                    else
+                        result = BigCalc.Multiply(left._bits, right._bits);
+                }
+                else
```

Comment:
Really? It's a bit verbose though... Or only the the `else` statement without braces of the `if` statement with braces?